practice/tic-tac-toe/dict_board.py

```diff
@@ -0,0 +1,192 @@
+
+
+class DictTTTBoard:
+    """Creates a Board class as a dictionary of coordinates to tokens."""
+
+    def __init__(self):
+        r"""Initialize Class
+        >>> a = DictTTTBoard()
+        >>> sorted(a._coord_to_token)
+        [(0, 0), (0, 1), (0, 2), (1, 0), (1, 1), (1, 2), (2, 0), (2, 1), (2, 2)]
+        """
+        self._coord_to_token = {
+            (0,0): ' ', (1,0): ' ', (2,0): ' ',
+            (0,1): ' ', (1,1): ' ', (2,1): ' ',
+            (0,2): ' ', (1,2): ' ', (2,2): ' '
+        }
+
+    def __repr__(self):
+        """Return representation.
+        # concerned with how useless repr will be for human interpretation.
+
+        >>> a = DictTTTBoard()
+        >>> a._coord_to_token = {(0, 0): 'X'}
+        >>> a
+        DictTTTBoard({(0, 0): 'X'})
+        """
+        return 'DictTTTBoard({!r})'.format(self._coord_to_token)
+
+    def __eq__(self, other):
+        """Defines equality.
+
+        >>> a = DictTTTBoard()
+        >>> a._coord_to_token = {(0, 0): 'X'}
+        >>> b = DictTTTBoard()
+        >>> b._coord_to_token = {(0, 0): 'X'}
+        >>> a == b
+        True
+        >>> a = DictTTTBoard()
+        >>> a._coord_to_token = {(0, 0): 'X'}
+        >>> b = DictTTTBoard()
+        >>> b._coord_to_token = {(0, 1): 'X'}
+        >>> a == b
+        False
+        """
+        return (self._coord_to_token == other._coord_to_token)
+
+    def place_token(self, x, y, token):
+        """Place a token character string at a given coordinate.
+
+        (top-left is 0, 0), x is horizontal position, y is vertical position
+
+        >>> a = DictTTTBoard()
+        >>> a._coord_to_token = {(0, 0): 'X'}
+        >>> a.place_token(0, 1, 'O')
+        >>> b = a._coord_to_token
+        >>> sorted(b.items())
+        [((0, 0), 'X'), ((0, 1), 'O')]
+        >>> a = DictTTTBoard()
+        >>> a._coord_to_token = {(0, 0): 'X', (0,1): 'O'}
+        >>> a.place_token(0, 2, 'X')
+        >>> b = a._coord_to_token
+        >>> sorted(b.items())
+        [((0, 0), 'X'), ((0, 1), 'O'), ((0, 2), 'X')]
+        """
+        self._coord_to_token[(x, y)] = token
+
+    def make_rows(self):
+        """Returns a list of listed tokens, in order of the rows going top down.
+
+        >>> a = DictTTTBoard()
+        >>> a._coord_to_token = {
+        ...   (0,0): 'X', (1,0): ' ', (2,0): 'O',
+        ...   (0,1): 'X', (1,1): 'O', (2,1): 'X',
+        ...   (0,2): 'O', (1,2): ' ', (2,2): 'X'
+        ...    }
+        >>> a.make_rows()
+        [['X', ' ', 'O'], ['X', 'O', 'X'], ['O', ' ', 'X']]
+        """
+        coords_to_tokens = sorted(list(self._coord_to_token.items()))
+        row_1 = [coord_to_token[1] for coord_to_token in coords_to_tokens if coord_to_token[0][1] == 0]
+        row_2 = [coord_to_token[1] for coord_to_token in coords_to_tokens if coord_to_token[0][1] == 1]
+        row_3 = [coord_to_token[1] for coord_to_token in coords_to_tokens if coord_to_token[0][1] == 2]
+        return [row_1, row_2, row_3]
```

Comment:
I know this code is repetitive, but I'm really struggling to figure out how to put a line comprehension into a line comprehension, if that makes sense.